src/v2/components/settings/SettingsModal.tsx

```
@@ -0,0 +1,160 @@
+  };
+
+  const ActiveComponent =
+    tabs.find((tab) => tab.id === activeTab)?.component || tabs[0].component;
+
+  return (
+    <div
+      className="fixed inset-0 z-50 bg-black/50 flex items-center justify-center p-4"
+      onClick={handleContentClick}
+      onKeyDown={handleContentKeyDown}
+      role="dialog"
+    >
```

Biome (1.9.4)

[error] The elements with the following roles can be changed to the following elements:
<dialog>


For examples and more information, see WAI-ARIA Roles

(lint/a11y/useSemanticElements)